File: tasks/implementations/java/tasks/cli_file_search.java
import java.nio.file.*;
import java.util.stream.*;

class CliFileSearch {
    public static void run(String size, String fixturesRoot) throws Exception {
        long total = 0;
        try (Stream<Path> paths = Files.walk(Path.of(fixturesRoot, "generated", "search", size))) {
            for (Path path : (Iterable<Path>) paths.filter(Files::isRegularFile)::iterator) {
                if (path.toString().endsWith(".txt")) {
                    for (String line : Files.readAllLines(path)) {
                        if (line.contains("needle")) total += 1;
                    }
                }
            }
        }
        System.out.println(total);
    }
}

File: tasks/implementations/node/tasks/cli_file_search.js
const fs = require('fs');
const path = require('path');
function walk(dir) {
  let total = 0;
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) total += walk(fullPath);
    else if (entry.isFile() && entry.name.endsWith('.txt')) {
      for (const line of fs.readFileSync(fullPath, 'utf8').split(/\r?\n/)) {
        if (line.includes('needle')) total += 1;
      }
    }
  }
  return total;
}
module.exports = function(size, fixturesRoot) {
  console.log(walk(`${fixturesRoot}/generated/search/${size}`));
};
